blog/2025-03-25-K3s-bootstrap.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "K3s bootstrap deep dive"
+description: Explain k3s embeds components and how it bootstraps
```

Comment:
"Explains how K3S embeds components and bootstraps" ?

Reply:
We use the term "bootstrap" for multiple things in k3s and rke2. Instead of just saying "bootstraps" I would recommend saying specifically what you are covering, and avoiding use of this overloaded term as much as possible.

At different points, we use the term to cover:
* Extracting binaries and chart/manifests to disk.
* Acquiring certs+keys and node configuration such that the kubelet and kube-proxy can be started.
* Obtaining a copy of the cluster CA certs+keys, placing those on disk, and starting etcd and the apiserver.